### PR TITLE
Add missing InstrumenterBuilder.addRequestListener()

### LINKED ISSUE #5652: requestListeners In InstrumenterBuilder
How to add requestListener to InstrumenterBuilder,I can not find it in version 1.11.x

### PR DESCRIPTION
**Pull request by trask**
Resolves #5652

### CODE REVIEW

**Review by laurit — APPROVED** [review]

**Review by anuraaga — APPROVED** [review]

File: instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/Instrumenter.java [hunk 1]
```diff
@@ -232,10 +232,11 @@ public void end(
 
     if (!requestListeners.isEmpty() || !requestMetricListeners.isEmpty()) {
       long endNanos = getNanos(endTime);
-      for (RequestListener requestListener : requestListeners) {
+      // TODO (trask) call end in the reverse order that start was called?
```
Comment (mateuszrzeszutek) [inline instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/Instrumenter.java:235]: We probably don't have to do that - unlike `start()`, the `end()` method cannot modify the context; also, we're passing the timestamp as an argument, so the exact time of invocation should not matter too.

File: instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/Instrumenter.java [hunk 2]
```diff
@@ -232,10 +232,11 @@ public void end(
...
+      for (RequestListener requestListener : requestMetricListeners) {
         requestListener.end(context, attributes, endNanos);
       }
-      for (RequestListener requestListener : requestMetricListeners) {
+      for (RequestListener requestListener : requestListeners) {
```
Comment (mateuszrzeszutek) [inline instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/Instrumenter.java:239]: Nit: do we actually need two lists of listeners? Can't we merge `requestMetricListeners` into `requestListeners`?

**Review by mateuszrzeszutek — APPROVED** [review]

File: instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/Instrumenter.java (hunk 1 above)
Comment (trask) [reply instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/Instrumenter.java:235]: ya, I just always think of start/end pairs telescoping :)

File: instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/Instrumenter.java (hunk 2 above)
Comment (trask) [reply instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/Instrumenter.java:239]: I'll send that in a separate PR b/c it moves the requestListeners to after the span is started (which I think is ok, but I'd like to get eyes specifically on that change)

**Pull request merged**